reporters/reporter-default/src/main/java/org/radargun/reporting/html/ReportDocument.java

```diff
@@ -573,36 +397,135 @@ protected void writeScripts() {
+              clusterSize, operation, "Operations/sec", ChartType.OPERATION_THROUGHPUT_NET))
+         generatedGraphs.add("throughput_net");
+      if (createChart(
+              String.format("%s%s%s_%s%s_data_throughput.png", directory, File.separator, testName, operation, suffix),
+              clusterSize, operation, "MB/sec", ChartType.DATA_THROUGHPUT))
+         generatedGraphs.add("data_throughput");
+   }
 
-      @Property(doc = "Highlight suspicious results in the report. Default is true.")
-      protected boolean highlightSuspects = true;
+   public OperationData getOperationData(final String operation, Map<Report, List<Aggregation>> reportAggregationMap) {
+
+      Collection<StatisticType> presentedStatistics = new ArrayList<>();
+
+      presentedStatistics.add(StatisticType.MEAN_AND_DEV);
+      if (configuration.percentiles.length > 0 && hasRepresentation(operation, reportAggregationMap, Percentile.class, configuration.percentiles[0])) {
+         presentedStatistics.add(StatisticType.PERCENTILES);
+      }
+      if (hasRepresentation(operation, reportAggregationMap, Histogram.class, configuration.histogramBuckets, configuration.histogramPercentile)) {
+         presentedStatistics.add(StatisticType.HISTOGRAM);
+      }
+      if (hasRepresentation(operation, reportAggregationMap, OperationThroughput.class, 1L)) {
+         presentedStatistics.add(StatisticType.OPERATION_THROUGHPUT);
+      }
+      if (hasRepresentation(operation, reportAggregationMap, DataThroughput.class, 100L, 100L, 100L, 100.0)) {
+         presentedStatistics.add(StatisticType.DATA_THROUGHPUT);
+      }
+
+      List<String> iterations = new ArrayList<>(maxIterations);
+      for (int iteration = 0; iteration < maxIterations; ++iteration) {
+         // in fact we shouldn't have different iterations values for iterations with the same id,
+         // but it's possible
+         Set<String> iterationValues = new HashSet<>();
+         for (List<Aggregation> aggregations : reportAggregationMap.values()) {
+            if (aggregations != null && iteration < aggregations.size()) {
+               Aggregation aggregation = aggregations.get(iteration);
+               if (aggregation != null && aggregation.iteration.getValue() != null) {
+                  iterationValues.add(aggregation.iteration.test.iterationsName + " = " + aggregation.iteration.getValue());
+               }
+            }
+         }
+         iterations.add(concatOrDefault(iterationValues, "iteration " + String.valueOf(iteration)));
+      }
+
+      for (Future f : chartTaskFutures) {
+         try {
+            f.get();
+         } catch (Exception e) {
+            log.error("Failed to create chart", e);
+         }
+      }
+      chartTaskFutures.clear();
+
+      return new OperationData(presentedStatistics, iterations);
+   }
+
+   //These methods are needed in testReport template
+   public abstract List<TestAggregations> getTestAggregations();
```

Comment:
Are you sure you need those abstract methods here? Wouldn't it be better to declare those methods in subclasses as necessary?